### PR TITLE
add cloudbees dev. team to OSS plugins that CB  (co-)maintains

### PR DESCRIPTION
**Pull request by mikecirioli**
# Description

Adds [Cloudbees Developers](https://github.com/orgs/jenkinsci/teams/company-cloudbees-developers/members) to the list of developers for plugins that Cloudbees (co-)maintains for easier management.

The following repositories are affected:

The checkbox means the repo got the necessary approval from one or more existing releaser/maintainer.

- [x] https://github.com/jenkinsci/artifact-manager-s3-plugin approved by @jglick 
- [x] https://github.com/jenkinsci/aws-credentials-plugin approved by @escoem
- [x] https://github.com/jenkinsci/aws-global-configuration-plugin  approved by @jglick 
- [x] https://github.com/jenkinsci/aws-java-sdk-plugin approved by @Vlatombe
- [x] https://github.com/jenkinsci/azure-publishersettings-credentials-plugin approved by @batmat
- [x] https://github.com/jenkinsci/kubernetes-client-api-plugin/ approved by @Vlatombe @jglick 
- [x] https://github.com/jenkinsci/kubernetes-credentials-plugin/ approved by @maxlaverse 
- [x] https://github.com/jenkinsci/kubernetes-plugin  approved by @Vlatombe @jglick 
- [x] https://github.com/jenkinsci/mapdb-api-plugin approved by @stephenc
- [x] https://github.com/jenkinsci/view-job-filters-plugin approved by @jglick 

FYI to current maintainers of these plugins:  @jglick @oleg-nenashev  @kuisathaverat @svenschoenung @maxlaverse @cronik @timja @andresrc 

# Submitter checklist for adding or changing permissions

### Always

- [x] Add link to plugin/component Git repository in description above

### For a newly hosted plugin only

- [ ] Add link to resolved HOSTING issue in description above

### For a new permissions file only

- [ ] Make sure the file is created in `permissions/` directory
- [ ] `artifactId` (pom.xml) is used for `name` (permissions YAML file).
- [ ] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)
- [ ] Check that the file is named `plugin-${artifactId}.yml` for plugins

### When adding new uploaders (this includes newly created permissions files)

- [ ] [Make sure to `@`mention an existing maintainer to confirm the permissions request, if applicable](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)
- [ ] Use the Jenkins community (LDAP) account name in the YAML file, not the GitHub account name
- [ ] Make sure to `@`mention the users being added so their GitHub account names are known if they require GitHub merge access (see below).
- [ ] [All newly added users have logged in to Artifactory at least once](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)

### Reviewer checklist (not for requesters!)

- [ ] Check this if newly added person also needs to be given merge permission to the GitHub repo (please @ the people/person with their GitHub username in this issue as well). If needed, it can be done using an [IRC Bot command](https://jenkins.io/projects/infrastructure/ircbot/#github-repo-management)
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@daniel-beck`) in this pull request. If an email contact is changed, wait for approval from the security officer.

There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it

### CODE REVIEW

File: permissions/plugin-azure-keyvault.yml
Comment (timja) [inline permissions/plugin-azure-keyvault.yml:12]: ? this isn't cloudbees maintained

Comment (mikecirioli) [reply permissions/plugin-azure-keyvault.yml]: oops, yes - i mistakenly added the group to the wrong plugin, change reverted and correct plugin updated

**Review by jglick — COMMENTED** [review]
OK for me for the plugins I am already listed next to.

File: permissions/plugin-azure-publishersettings-credentials.yml
Comment (timja) [inline permissions/plugin-azure-publishersettings-credentials.yml:11]: this plugin is dead ftr, it was emptied out: https://github.com/jenkinsci/azure-publishersettings-credentials-plugin

**Comment by timja** [comment]
Is cloudbees developers an actual account?

**Review by jtnord — CHANGES_REQUESTED** [review]
This is not supported (yet) give it a week or two and we should be able to do something like this bit until then RPU has no support for groups.

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Needs explicit approvals from non-CloudBees maintainers listed for plugins

File: permissions/plugin-saml.yml
Comment (kuisathaverat) [inline permissions/plugin-saml.yml:14]: SAML Plugin is not maintained by CloudBees

```suggestion
```

**Review by kuisathaverat — CHANGES_REQUESTED** [review]
SAML Plugin is not maintained by CloudBees

**Comment by maxlaverse** [comment]
Fine with me!

File: permissions/plugin-saml.yml
Comment (kuisathaverat) [inline permissions/plugin-saml.yml:14]:
```suggestion
```

**Comment by timja** [comment]
conflicts

**Review by kuisathaverat — APPROVED** [review]

**Comment by batmat** [comment]
I think for some plugins like mapdb we may need to officially ask for adopting it, given there's no current declared releaser. :thinking:

**Comment by timja** [comment]
> I think for some plugins like mapdb we may need to officially ask for adopting it, given there's no current declared releaser. 🤔

mapdb appears to be @stephenc

**Comment by stephenc** [comment]
Oh mapdb should be handed over to someone else in cloudbees. I should only be left as maintaining gitea (until I get the one thing I’m trying to finish on it done)

**Comment by timja** [comment]
@batmat are you able to make a checklist against each plugin and check approvals please?

**Comment by batmat** [comment]
@timja yeah sure, we'll do it. Thanks

**Comment by escoem** [comment]
I'm ok for the aws-credentials plugin https://github.com/jenkins-infra/repository-permissions-updater/blob/master/permissions/plugin-aws-credentials.yml#L12

**Comment by Vlatombe** [comment]
ok for me for
* kubernetes
* kubernetes-client-api
* aws-java-sdk
* aws-credentials

**Comment by batmat** [comment]
@timja I've created and updated the table. We're complete AFAICT, we have an approval of 1+ person per plugin.

@oleg-nenashev I assume your request was on saml plugin mainly. We've not generally asked for an approval from _all_ releasers till now?

Or are you explicitly rejecting this PR for some of the plugins you're declared for? Thanks!

**Comment by jtnord** [comment]
ok for me (which is only `jenkinsci/kubernetes-plugin` where I currently have deploy / maintainer permissions)

**Comment by jtnord** [comment]
I asked @stephenc to add  `jenkinsci/company-cloudbees-developers` to the [jenkinsci/mapdb-api-plugin](https://github.com/jenkinsci/mapdb-api-plugin) repository with admin access to complete the handover but he has no access.  As per the approval above from Stephen above can we take this as an green light and can someone with the org admin can make the change when this PR is merged?

**Comment by timja** [comment]
You should be able to add yourself via the bot and then sort it

**Comment by batmat** [comment]
@timja do you think the bot will be able to add a team, interesting, we should try :)

**Comment by timja** [comment]
> @timja do you think the bot will be able to add a team, interesting, we should try :)

err i just meant adding the team to the repo. and James could give himself admin permissions

**Comment by batmat** [comment]
@oleg-nenashev can you please answer my question? Thanks!

File: permissions/plugin-artifact-manager-s3.yml
Comment (timja) [inline permissions/plugin-artifact-manager-s3.yml:12]: > Team company-cloudbees-developers not found

the team is called `cloudbees-developers`
https://github.com/jenkins-infra/repository-permissions-updater/blob/master/teams/cloudbees-developers.yml

Comment (batmat) [reply permissions/plugin-artifact-manager-s3.yml:12]: Yes my bad. I saw it the other day, but I left this fixing on the side as I wanted to get Oleg's clarification most importantly.
I'll fix it.

Comment (batmat) [reply permissions/plugin-artifact-manager-s3.yml:12]:
```suggestion
- "@cloudbees-developers"
```

Comment (timja) [reply permissions/plugin-artifact-manager-s3.yml]: It’s in all the files…

File: permissions/plugin-aws-credentials.yml
Comment (batmat) [inline permissions/plugin-aws-credentials.yml:15]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-aws-global-configuration.yml
Comment (batmat) [inline permissions/plugin-aws-global-configuration.yml:13]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-aws-java-sdk.yml
Comment (batmat) [inline permissions/plugin-aws-java-sdk.yml:14]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-kubernetes-client-api.yml
Comment (batmat) [inline permissions/plugin-kubernetes-client-api.yml:17]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-kubernetes-credentials.yml
Comment (batmat) [inline permissions/plugin-kubernetes-credentials.yml:12]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-kubernetes.yml
Comment (batmat) [inline permissions/plugin-kubernetes.yml:13]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-mapdb-api.yml
Comment (batmat) [inline permissions/plugin-mapdb-api.yml:9]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-view-job-filters.yml
Comment (batmat) [inline permissions/plugin-view-job-filters.yml:12]:
```suggestion
- "@cloudbees-developers"
```

File: permissions/plugin-azure-publishersettings-credentials.yml
Comment (batmat) [inline permissions/plugin-azure-publishersettings-credentials.yml:11]:
```suggestion
- "@cloudbees-developers"
```

**Pull request merged**
